src/mongoc/mongoc-topology-scanner.c

```diff
@@ -398,13 +399,18 @@ mongoc_topology_scanner_start (mongoc_topology_scanner_t *ts,
 
    DL_FOREACH_SAFE (ts->nodes, node, tmp)
    {
-      if (mongoc_topology_scanner_node_setup (node)) {
+      /* FIXME: Gotta catch all the pokemons, not overwrite them */
+      bson_error_t error;
```

Comment:
I agree we should try to resolve the fundamental error to human digestible error message -- I'm a littlebit worried though that we may be hiding away the gory details.

It seems like if there is any connectible node then pymongo doesn't tell you why there isn't a matching node. It just tells you that there wasn't a matching node for the 'selector'.

I guess there is no way to correlate server errors to the "selector", but we could attach the server errors as a "full details" option somehow?
Not sure how to handle that though, but I was really hoping to be able to provide transport error messages :/
We should at least make sure they are logged.